wicker/core/storage.py

```diff
@@ -34,7 +35,7 @@ class AbstractDataStorage(ABC):
 
     @abstractmethod
     def fetch_file(self, input_path: str, local_prefix: str, timeout_seconds: int) -> str:
-        """Fetch file from chosen data storage method.
+        """Fetch file from data storage into the local path.
```

Comment:
Do we still have the concept of fetch for the mounted Fuse DataStorage? I am assuming the end user just care about the content, Do we need make this ensure path method as a public interface?

Reply:
input_path-> storage_path, add please add doc string for what is the input path. Wether relative path to each DataStorage, or abs path. I am assuming the relative path since the DataStorage class would be a holder for all the interactive with file based interactive

Reply:
@zhenyu the way that `S3DataStorage` was created is that unfortunately `input_path` represents an absolute path within that type of storage, for example the code looks like this:
```
data_storage = S3DataStorage()
input_path = "s3://foo/bar/baz/dummy"
local_prefix = "/tmp/data"
data_storage.fetch_file(input_path, local_prefix)
```
I think it would have been better to have made it something like:
```
data_storage = S3DataStorage(root_path="s3://foo/bar")
input_path = "baz/dummy"
local_prefix = "/tmp/data"
data_storage.fetch_file(input_path, local_prefix)
```
But for backwards compatibility I think it is too late to change this for `S3DataStorage`. However, we could change it for `FileSystemDataStorage`.

What do you think, in the `AbstractDataStorage` interface should we specify that `input_path` must be an absolute path in storage (so that we match the behavior of `S3DataStorage`) or should we leave it up to the derived class like `FileSystemDataStorage` to decide on whether `input_path` is relative or absolute path???